modules/opensimplex/open_simplex_noise.cpp

```diff
@@ -96,15 +96,15 @@ void OpenSimplexNoise::set_lacunarity(float p_lacunarity) {
 	emit_changed();
 }
 
-Ref<Image> OpenSimplexNoise::get_image(int p_width, int p_height) const {
+Ref<Image> OpenSimplexNoise::get_image(int p_width, int p_height, int p_offset_x, int p_offset_y) const {
 	Vector<uint8_t> data;
 	data.resize(p_width * p_height);
 
 	uint8_t *wd8 = data.ptrw();
 
 	for (int i = 0; i < p_height; i++) {
 		for (int j = 0; j < p_width; j++) {
-			float v = get_noise_2d(j, i);
+			float v = get_noise_2d(j + p_offset_x, i + p_offset_y);
```

Comment:
This is the meat of the enhancement. Everything else is plumbing.

Reply:
`get_noise_2d` functions accept `float` for x/y parameters, but `int p_offset_x, int p_offset_y` in `get_image()` accept integers. Would this be limited to integer offsets? Would it make any difference to use float offset over integer offset?

Reply:
I used `int` because the existing operation iterates over `int` values. What you are suggesting makes sense too, though. Do you think it would be valuable to have this accept `float` instead?

Reply:
Yeah, the loop operates on `int` values because noise is mapped to image coordinates, yet noise is not discrete compared to image, so it makes sense to me for this to accept `float` (rather, if we're talking about offset, perhaps it should use `real_t`, which is `double` in Godot by default, especially when we're talking about procedural generation with infinite worlds).

You could also change it to accept `Vector2`, there are many similar `offset` properties throughout Godot which are `Vector2`s, and because you're already using `Vector2` for `noise_offset` property in `NoiseTexture`, which would make the API consistent.

I'm not 100% sure about this, @JFonS may know better.

Reply:
Great idea. Thanks for the feedback. I've updated the PR to use `Vector2` as the type throughout.